### PR TITLE
Use alias to report container image in k8s parts

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR leverages `type :alias`  for `kubernetes.container.image` so as to copy the fields under ECS `container.image.name`.
This affects metadata collected by kubernetes autodiscover provider, `add_kubernetes_metadata` processor and `state_container` metricset.

## Why is it important?
In order to have a persistent way to report ECS container fields, without the need to duplicate fields and store both which will impact ECS.

## How to test this PR locally
1. Before all test verify that `fields.yml` on your testing env is properly updated with the `alias` change and manually `setup` and verify that mappings are properly loaded in ES.
2. Check that autodiscover meta work:
```
filebeat.autodiscover:
      providers:
        - type: kubernetes
          node: ${NODE_NAME}
          hints.enabled: true
          hints.default_config:
            type: container
            paths:
              - /var/log/containers/*${data.kubernetes.container.id}.log
```
Make sure that `container.image.name` exist in the final documents in ES and you able to search on based on it while  `kubernetes.container.image` is not populated in the event but is searchable because of the `alias`.
3. Check that `add_kubernetes_metadata` processor works:
```
filebeat.inputs:
    - type: container
      paths:
        - /var/log/containers/*.log
      processors:
        - add_kubernetes_metadata:
            host: ${NODE_NAME}
            indexers:
            - container:
            matchers:
            - logs_path:
                logs_path: "/var/log/containers/"
```
Make sure that `container.image.name` exist in the final documents in ES and you able to search on based on it while  `kubernetes.container.image` is not populated in the event but is searchable because of the `alias`.

4. Verify with `state_container` metricset enabled that, `container.image.name` exist in the final documents in ES and you able to search based on it while  `kubernetes.container.image` is not populated in the event but is searchable because of the `alias`.

## Related issues

- Closes https://github.com/elastic/beats/issues/23585
- Relates https://github.com/elastic/beats/pull/20984, https://github.com/elastic/beats/pull/23802

## Screenshots
Screenshot 2021-03-10 at 12 30 38 PM
Screenshot 2021-03-10 at 12 30 28 PM
Screenshot 2021-03-10 at 12 41 07 PM
Screenshot 2021-03-10 at 12 40 54 PM

### CODE REVIEW

File: libbeat/processors/add_kubernetes_metadata/_meta/fields.yml [hunk 1]
```diff
@@ -70,11 +70,13 @@
             Kubernetes statefulset name
 
         - name: container.name
-          type: keyword
+          type: alias
+          path: container.name
```
Comment (kaiyan-sheng) [inline libbeat/processors/add_kubernetes_metadata/_meta/fields.yml:74]: Is this intentional?

Comment (ChrsMark) [reply libbeat/processors/add_kubernetes_metadata/_meta/fields.yml:74]: We want `kubernetes.container.name` to point to `container.name` (same for image) so yes?

Comment (jsoriano) [reply libbeat/processors/add_kubernetes_metadata/_meta/fields.yml:74]: Yes, full paths need to be configred here, I think this is correct.

File: libbeat/autodiscover/providers/kubernetes/pod.go [hunk 2]
```diff
@@ -356,10 +356,7 @@ func (p *pod) emitEvents(pod *kubernetes.Pod, flag string, containers []kubernet
 		)
 
 		cmeta := common.MapStr{
-			"id": cid,
-			"image": common.MapStr{
-				"name": c.Image,
-			},
```
Comment (jsoriano) [inline libbeat/autodiscover/providers/kubernetes/pod.go:362]: This seems to be removing fields from the `container` object in meta, shouldn't we be removing the fields from `kubemeta["container"]`, so they are removed from `kubernetes.container` object in meta?

Comment (ChrsMark) [reply libbeat/autodiscover/providers/kubernetes/pod.go:362]: well If i'm not mistaken `cmeta` is the final object that will be used to enrich the events so we explicitly add `id` and `runtime` here while the container.image.name and container.name will be added through the alias.

`kubemeta` on the other hand contains the data that will be used to evaluate configurations like in a condition: 
```
contains: kubernetes.container.image: redis
```

Does this make sense?

File: libbeat/processors/add_kubernetes_metadata/_meta/fields.yml (hunk 1 above)
Comment (kaiyan-sheng) [reply libbeat/processors/add_kubernetes_metadata/_meta/fields.yml:74]: Ahh yep sorry I missed it! Thanks!!

File: libbeat/autodiscover/providers/kubernetes/pod.go (hunk 2 above)
Comment (jsoriano) [reply libbeat/autodiscover/providers/kubernetes/pod.go:362]: Oh ok, makes sense, I always get confused about these metas.

**Comment by ChrsMark** [comment]
@jsoriano @kaiyan-sheng could you please take another look please? https://github.com/elastic/beats/pull/24380#discussion_r588276592 seems to be valid since code should set the values under ECS fields and not under `kubernetes.container.*`. `kubernetes.container.*` will be populated through the `alias`. I also added testing notes and some screenshots.

**Review by jsoriano — COMMENTED** [review]
A couple of minor comments but it looks good.

File: metricbeat/module/kubernetes/state_container/_meta/data.json [hunk 3]
```diff
@@ -62,11 +60,15 @@
   },
   "container": {
     "runtime": "docker",
-    "id": "5f8ce416d10ab0b28ce5c7d521de2264aa03ff4d001e1194076f6a02a330139f"
+    "id": "5f8ce416d10ab0b28ce5c7d521de2264aa03ff4d001e1194076f6a02a330139f",
+    "image": {
+      "name": "ubuntu:latest"
+    },
+    "name": "ubuntu"
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply metricbeat/module/kubernetes/state_container/_meta/data.json:67]: On the other hand, the image reported by `kubernetes.container.image` is the image in the runtime, so it is probably ok if this is moved to a common field.

**Review by jsoriano — APPROVED** [review]
Thanks!

**Pull request merged**